src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,156 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Dao
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?,?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement
+                        = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+
+            int affectedRows = preparedStatement.executeUpdate();
+
+            try {
+                boolean b = affectedRows < 1;
+            } catch (DataProcessingException ex) {
+                throw new DataProcessingException("Expected to insert at least one row, "
+                        + "but inserted 0 rows.", ex);
+            }

```

Comment:
The try-catch block here is incorrectly used. The condition `boolean b = affectedRows < 1;` does not throw an exception, so catching a `DataProcessingException` here is not appropriate. Instead, you should directly check if `affectedRows < 1` and throw the exception if true.

Reply:
ok, done